src/pages-and-resources/teams/Settings.jsx

```diff
@@ -0,0 +1,105 @@
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button, Icon } from '@edx/paragon';
+import { Add } from '@edx/paragon/icons';
+import { FieldArray } from 'formik';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { v4 as uuid } from 'uuid';
+import * as Yup from 'yup';
+import { useAppSetting } from '../../utils';
+import AppSettingsModal from '../app-settings-modal/AppSettingsModal';
+import messages from './messages';
+import TeamSetEditor, { TeamSetTypes, TeamSizes } from './TeamSetEditor';
+
+function TeamSettings({
+  intl,
+  onClose,
+}) {
+  const [teamsConfiguration, saveSettings] = useAppSetting('teamsConfiguration');
+  const blankNewTopic = {
+    name: '',
+    description: '',
+    type: 'open',
+    maxTeamSize: 5,
+    id: null,
+  };
+
+  const handleSettingsSave = (values) => {
+    // For newly-added teams, fill in an id.
+    const topics = values.topics.map(topic => ({
+      id: topic.id || uuid(),
```

Comment:
Instead of a UUID perhaps it might be better to slugify the topic name? This will result in friendlier URLs.

Reply:
@xitij2000 

I don't think it makes much of a difference overall. But perhaps it wouldn't be a bad idea either. I'd rather the slug most probably.

Reply:
The difference it makes is that this ID is used in URLs. So in one case users see a URL like '/north-america' and in the other '/6432aabb-0829-4ca5-b9b5-83829c94f277'.

Reply:
Actually this might be harder said than done, especially if you consider unicode. I think the best approach is a uuid or a manual slug.